src/test/java/uk/gov/companieshouse/limitedpartnershipsapi/service/PostTransitionPartnerTest.java

```diff
@@ -246,6 +246,64 @@ void shouldReturn200AndNoFeeForKind() throws Exception {
     }
 
+    @Nested
+    class RemoveLimitedPartnerLegalEntity {
+        @Test
+        void shouldReturn200IfNoErrors() throws Exception {
+            CompanyProfileApi companyProfile = new CompanyBuilder().build();
+            when(companyService.getCompanyProfile(any())).thenReturn(companyProfile);
+
+            limitedPartnerLegalEntityDao.getData().setCeaseDate(LocalDate.of(2025, 1, 1));
+            limitedPartnerLegalEntityDao.getData().setRemoveConfirmationChecked(true);
+
+            mocks(PartnerKind.REMOVE_LIMITED_PARTNER_LEGAL_ENTITY, generalPartnerLegalEntityDao, limitedPartnerLegalEntityDao);
+
+            var result = limitedPartnerService.validateLimitedPartner(transactionLimitedPartner, limitedPartnerLegalEntityDao.getId());
+
+            assertThat(result).isEmpty();
+        }
+
+        @Test
+        void shouldReturn200AndErrorDetailsIfNoCeaseDate() throws Exception {
+            CompanyProfileApi companyProfile = new CompanyBuilder().build();
+            when(companyService.getCompanyProfile(any())).thenReturn(companyProfile);
+
+            mocks(PartnerKind.REMOVE_LIMITED_PARTNER_LEGAL_ENTITY, generalPartnerLegalEntityDao, limitedPartnerLegalEntityDao);
+
+            limitedPartnerLegalEntityDao.getData().setCeaseDate(null);
+            limitedPartnerLegalEntityDao.getData().setRemoveConfirmationChecked(false);
+
+            var result = limitedPartnerService.validateLimitedPartner(transactionLimitedPartner, limitedPartnerLegalEntityDao.getId());
+
+            assertThat(result).hasSize(2)
+                    .extracting(e -> Map.entry(e.getLocation(), e.getError()))
+                    .containsExactlyInAnyOrder(
+                            Map.entry("data.ceaseDate", "Cease date is required"),
+                            Map.entry("data.removeConfirmationChecked", "Remove confirmation checked is required")
+                    );
+        }
+
+        @Test
+        void shouldReturn200AndErrorDetailsIfCeaseDateBeforeIncorporationDate() throws Exception {
+
+            CompanyProfileApi companyProfile = new CompanyBuilder().build();
+            when(companyService.getCompanyProfile(any())).thenReturn(companyProfile);
+
+            limitedPartnerLegalEntityDao.getData().setCeaseDate(LocalDate.of(2000, 1, 1));
+            limitedPartnerLegalEntityDao.getData().setRemoveConfirmationChecked(true);
+
+            mocks(PartnerKind.REMOVE_LIMITED_PARTNER_LEGAL_ENTITY, generalPartnerLegalEntityDao, limitedPartnerLegalEntityDao);
+
+            var result = limitedPartnerService.validateLimitedPartner(transactionLimitedPartner, limitedPartnerLegalEntityDao.getId());
+
+            assertThat(result).hasSize(1)
+                    .extracting(e -> Map.entry(e.getLocation(), e.getError()))
+                    .containsExactlyInAnyOrder(
+                            Map.entry("data.ceaseDate", "Partner cease date cannot be before the incorporation date")
+                    );
+        }
+    }
+
```

Comment:
you should also be able to add the "shouldReturn200AndNoFeeForKind" test here , see the one i added for remove LP person

Reply:
Introduced this new test